substrate/frame/revive/src/exec.rs

```diff
@@ -280,6 +280,11 @@ pub trait PrecompileWithInfoExt: PrecompileExt {
 pub trait PrecompileExt: sealing::Sealed {
 	type T: Config;
 
+	/// Charges the gas meter with the given weight.
+	fn charge(&mut self, weight: Weight) -> Result<crate::gas::ChargedAmount, DispatchError> {
+		self.gas_meter_mut().charge(crate::RuntimeCosts::Precompile(weight))
+	}
+
```

Comment:
note we could move gas_meter_mut to Ext, but then we would break  PrimitivePrecompile that needs it